### DISCUSSION TITLE
How Do I Do Regex Match In Host?

### DISCUSSION THREAD

**Discussion by RoyAscendFS** [dotnet/yarp#2555] [Q&A] [answered]
In my config file I want to route all traffic to one cluster if the subdomain matches certain pattern. I know it's possible to do something like this:
`"Hosts": [ "*.mydomain.com" ]`
and if I have different domains, then the above rule will be sufficient. However my challenge is all the traffic will have the same domain, it's only the subdomain that will be different.
The pattern I want to match is something like:
eg1-subdomain.mydomain.com
eg2-subdomain.mydomain.com
The pattern I don't want to match would be something like
nomatch.mydomain.com
so I've tried using this:
`"Hosts": [ "*-subdomain.mydomain.com" ]`
but it does not appear to do what I want, all the traffic is being send to the other cluster that catches all hosts.

Thanks in advance.

**Comment by benjaminpetit** [comment, marked as answer]
Unless I am mistaken, it seems to be a limitation of https://github.com/dotnet/aspnetcore/blob/main/src/Http/Routing/src/Matching/HostMatcherPolicy.cs
